### PR TITLE
feat: Add support for Route Import

### PR DESCRIPTION
**Pull request by parmus**
- feat: Add support for Route Import 
  This implementation of RouteImport() only supports the same subset of parameters as Routes()
- feat: Add support for "length" span attribute

### CODE REVIEW

File: routingv8/request.go
Comment (odsod) [inline routingv8/request.go:514]: Follow Go conventions and have the name being documented as the first word (or on the first line).

https://go.dev/blog/godoc

> The convention is simple: to document a type, variable, constant, function, or even a package, write a regular comment directly preceding its declaration, with no intervening blank line. Godoc will then present that comment as text alongside the item it documents. For example, this is the documentation for the fmt package’s [Fprint](https://go.dev/pkg/fmt/#Fprint) function:

```go
// Fprint formats using the default formats for its operands and writes to w.
// Spaces are added between operands when neither is a string.
// It returns the number of bytes written and any write error encountered.
func Fprint(w io.Writer, a ...interface{}) (n int, err error) {
```

> Notice this comment is a complete sentence that begins with the name of the element it describes. This important convention allows us to generate documentation in a variety of formats, from plain text to HTML to UNIX man pages, and makes it read better when tools truncate it for brevity, such as when they extract the first line or sentence.

**Review by odsod — APPROVED** [review]
LGTM, nice! See comment about GoDoc formatting before merge

File: routingv8/request.go
Comment (odsod) [inline routingv8/request.go:520]: nit: why not make this a time.Time and handle the marshaling to HERE maps format in the function itself? the caller might not know what format to use?

**Review by alethenorio — APPROVED** [review]
Looks good. A small nit

File: routingv8/request.go
Comment (parmus) [reply routingv8/request.go:520]: @odsod I agree with your reasoning, however this part was copied from the existing `RoutesRequest`, so I'm simply following the existing style. But I can also fix it. Which solution would you prefer?
1. Fixing it in both `RoutesRequest` and `RouteImportRequest`. This will keep the request types consistent, but will be a breaking change for any users of the library.
2. Fix just the `RouteImportRequest`, which won't break anything existing, but will make `RoutesRequest` and `RouteImportRequest` inconsistent.

File: routingv8/routes.go
Comment (parmus) [reply routingv8/routes.go:107]: @alethenorio I agree. However, this bit was actually copied from the existing `Routes()`. Do you want me to fix in both places?

File: routingv8/request.go
Comment (odsod) [reply routingv8/request.go:520]: Ahh thanks for the heads up - I'd say we don't need to block this PR on that then - if we make a breaking change might as well do it in 2 places!

I'm leaning towards saying let's do it and take approach nr.1 - but @alethenorio what do you think?

File: routingv8/routes.go
Comment (alethenorio) [reply routingv8/routes.go:107]: If you can be bothered that'd be nice :)

File: routingv8/request.go
Comment (alethenorio) [reply routingv8/request.go:520]: Yeah let's take a breaking change on this, I think that's OK given the size of the library and likely not a huge user base

Comment (alethenorio) [reply routingv8/request.go:520]: @parmus JUst mark the commit as breaking as well

File: routingv8/request.go
Comment (parmus) [reply routingv8/request.go:514]: Fixed

File: routingv8/routes.go
Comment (parmus) [reply routingv8/routes.go]: Fixed

File: routingv8/request.go
Comment (parmus) [reply routingv8/request.go:520]: @odsod @alethenorio Actually, I just realized that there's a reason for this: this parameter can take multiple different formats: https://www.here.com/docs/bundle/routing-api-v8-api-reference/page/index.html#tag/Routing/operation/calculateRoutes

> Specifies the time of departure as defined by either date-time or full-date T partial-time in RFC 3339, section 5.6 (for example, 2019-06-24T01:23:45).
>
> The requested time is converted to local time at origin. When the optional timezone offset is not specified, time is assumed to be local. The special value any can be used to indicate that time should not be taken into account during routing. If neither departureTime or arrivalTime are specified, current time at departure place will be used. All time values in the response are returned in the timezone of each location.
>
> Note: Only long-term traffic incidents will be used if departureTime=any and traffic[mode]=default or no traffic[mode] are specified.

So there are actually three case:
1) Time specified
2) Time not specified => current time of departure will be used
3) `"any"` specified which changes what Here Maps takes into account.

Thoughts?

Comment (parmus) [reply routingv8/request.go:520]: We could add a new type, which handles this logic, but it's starting to feel a bit out-of-scope for this PR. I'm happy to make a second separate PR to try and address this.

Comment (odsod) [reply routingv8/request.go:520]: spontaneous thoughts: probably separate struct fields with mutual exclusivity validation, or only support one format

Comment (alethenorio) [reply routingv8/request.go:520]: I'm thinking that it might make sense to keep as a string for now and take some time to think through how we want to change for something more robust. WDYT @odsod

Comment (parmus) [reply routingv8/request.go:520]: Sounds good to me as well. @odsod, any final thoughts on this?

Comment (parmus) [reply routingv8/request.go:520]: Cool. Then I'm keeping it as a string for now, and then we can address this in a later PR.

**Pull request merged**
